File: 0950-x-of-a-kind-in-a-deck-of-cards/solution.java

class Solution {
    public boolean hasGroupsSizeX(int[] deck) {
        Arrays.sort(deck);
        int temp = 0;
        int index =0;
        ArrayList<Integer> al = new ArrayList<Integer>();
        for (int i =0;i<deck.length ;i++){
            if(deck[i]==deck[index]){
                temp++;
            }
            else{
                al.add(temp);
                index=i;
                temp=1;
            
        }
        }
        al.add(temp);
        if (Collections.min(al)<2)
            return false;
           
        Boolean b  =true;
        for (int i=2 ; i<=Collections.min(al) ; i++){
         
            b= true;
            for (int j:al){
               
                if(j%i!=0){
                    
                    b=false;
                   
                    continue;
                }
                
                    
            }
            if (b==true)
            return true;
        }
        
        return false;
    }
}

File: 0950-x-of-a-kind-in-a-deck-of-cards/solution.py
class Solution:
    def hasGroupsSizeX(self, deck: List[int]) -> bool:
        col = collections.Counter(deck)
        values = list(col.values())
        values.sort()
        print(values)
        if values[0]<2:
            return False
        print("till here")
        i =2
        bol=True
        while i <=values[0]:
            bol =True
            for j in values:
                
                if j%i!=0:
                    bol=False
                    continue
                    
            i=i+1
            if bol==True:
                return True
        return False
        
